apps/web/src/state/info/queries/tokens/priceData.ts

```diff
@@ -49,7 +49,13 @@ const fetchTokenPriceData = async (
   }
   try {
     const blocks = await getBlocksFromTimestamps(timestamps, 'asc', 500, chainName)
-    if (!blocks || blocks.length === 0) {
+    const blocksLength = blocks?.length ?? 0
+    if (blocksLength > 0 && chainName === 'BSC' && !checkIsStableSwap()) {
+      const data = blocks[blocksLength - 1]
+      blocks[blocksLength - 1] = { timestamp: data.timestamp, number: data.number - 32 }
```

Comment:
Should we filter all blocks whose height is greater than `latest - 32`? Seems like now we are just changing the block number of the last block and the corresponding timestamp might be incorrect?